chia/server/node_discovery.py

```diff
@@ -182,22 +202,29 @@ async def _connect_to_peers(self, random) -> None:
 
                 # We don't know any address, connect to the introducer to get some.
                 size = await self.address_manager.size()
-                if size == 0 or empty_tables or first:
-                    first = False
+                if size == 0 or retry_introducers or introducer_attempts == 0:
                     try:
                         await asyncio.sleep(introducer_backoff)
                     except asyncio.CancelledError:
                         return None
-                    await self._introducer_client()
-                    # there's some delay between receiving the peers from the
-                    # introducer until they get incorporated to prevent this
-                    # loop for running one more time. Add this delay to ensure
-                    # that once we get peers, we stop contacting the introducer.
-                    try:
-                        await asyncio.sleep(5)
-                    except asyncio.CancelledError:
-                        return None
-                    empty_tables = False
+                    # Run dual between DNS servers and introducers. One time query DNS server,
+                    # next two times query the introducer.
+                    if introducer_attempts % 3 == 0 and len(self.dns_servers) > 0:
+                        dns_address = random.choice(self.dns_servers)
```

Comment:
it seems round-robin would be better here. if one DNS server is down or broken, we want to test the next one, never the same one again. We could still shuffle the DNS server list to make the order we try them in random.

I think a simple way of implementing round-robin would be to rotate the list for every call